File: lab5/1-seq.js
const seq = (x) => {
    const functions = []

    const addFunction = (x) => {
        if (typeof x === "function") {
            functions.push(x)
        }
        if (typeof x === "number") {
            functions.reverse()

            return functions.reduce((acc, currentValue) => currentValue(acc), x)
        }

        return addFunction
    }

    addFunction(x)
    return addFunction
}


const result = seq(x => x + 7)(x => x * 2)(5)

console.log(result)

File: lab5/1-seq.py
def seq(x):
    functions = []

    def add_function(x):
        if type(x) == 'function':
            functions.append(x)
        elif type(x) is int:
            functions.reverse()

            for i in functions:
                print(i)

        return add_function

    add_function(x)
    return add_function

def f1(x): return x * 2
def f2(x): return x + 1


result = seq(f1)(f2)(5)

print(result)